engine/minskyTensorOps.cc

```diff
@@ -276,9 +276,67 @@ namespace minsky
   {
     std::shared_ptr<ITensor> arg1, arg2;
     void computeTensor() const override {//TODO
-      throw runtime_error("inner product not yet implemented");
+		
+      if (arg1->rank()>2 || arg2->rank()>2)
+        throw runtime_error("inner product of rank>2 tensors not yet implemented");
+   		 
+      size_t stride=arg2->hypercube().dims()[0];	
+      size_t m=arg1->hypercube().dims()[0], n=arg2->hypercube().dims()[arg2->rank()-1];	 
```

Comment:
In general, m should be the product of dimensions 0..arg1->rank()-2, and n should be the product of dimensions 1..arg2->rank()-1

Reply:
That way, you'll be able to remove the test on line 280.